test/extensions/filters/network/thrift_proxy/conn_manager_test.cc

```diff
@@ -1925,6 +1936,71 @@ stat_prefix: test
   EXPECT_EQ(nullptr, route);
 }
 
+TEST_F(ThriftConnectionManagerTest, UnknownCluster) {
+  const std::string yaml = R"EOF(
+transport: FRAMED
+protocol: BINARY
+stat_prefix: test
+route_config:
+  name: "routes"
+  routes:
+    - match:
+        method_name: name
+      route:
+        cluster: cluster1
+    - match:
+        method_name: name2
+      route:
+        cluster: cluster2
+)EOF";
+
```

Comment:
nit: if you want, you can add back the assert for the case when the cluster list is empty.

Reply:
done. thanks!